src/java.desktop/share/classes/javax/sound/sampled/AudioSystem.java

```diff
@@ -238,17 +237,11 @@ public static Mixer getMixer(final Mixer.Info info) {
 
             currentInfoArray = mixer.getSourceLineInfo(info);
             for (int j = 0; j < currentInfoArray.length; j++) {
-                vector.addElement(currentInfoArray[j]);
+                list.add(currentInfoArray[j]);
             }
         }
 
-        Line.Info[] returnedArray = new Line.Info[vector.size()];
-
-        for (int i = 0; i < returnedArray.length; i++) {
-            returnedArray[i] = vector.get(i);
-        }
-
-        return returnedArray;
+        return list.toArray(new Line.Info[list.size()]);
```

Comment:
Shouldn't it be `new Line.Info[0]` as in the class below?

Reply:
It looks fine to change.